crypto/thread/task.c

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2019-2021 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/configuration.h>
+#include <openssl/crypto.h>
+
+#if defined(OPENSSL_THREADS)
+
+# include "task.h"
+
+static void crypto_task_free(CRYPTO_TASK *t);
+
+int task_list_cmp(struct list *l, void *data)
+{
+    return container_of(l, struct crypto_task_st, list)->id == *(uint32_t *)data;
+}
+
+CRYPTO_TASK *crypto_task_new(CRYPTO_THREAD_ROUTINE start, void *data)
+{
+    CRYPTO_TASK *t;
+
+    t = OPENSSL_zalloc(sizeof(*t));
+    if (t == NULL)
+        return NULL;
+
+    t->state = CRYPTO_TASK_NO_STATE;
+
+    t->lock = CRYPTO_MUTEX_create();
+    t->cond = CRYPTO_CONDVAR_create();
+
+    if (t->lock == NULL || t->cond == NULL)
+        goto fail;
+
+    if (CRYPTO_MUTEX_init(t->lock) == 0)
+        goto fail;
+
+    if (CRYPTO_CONDVAR_init(t->cond) == 0)
```

Comment:
The separate of `create` and `init` just looks odd. It's difficult to see when you would ever create without init, and what the point would be.

Reply:
The separate `create` and `init` were merged into `new`.